### PR TITLE
[ROCm] Forward host function names from gpu prefix to cuda prefix

### PR DESCRIPTION
**Pull request by jerryyin**
This PR is a follow-up to the original PR #28343. The reviewer requested to break down the original large PR to a series of smaller ones. According to the plan here, this PR is the 3rd one in the whole series. It is dependent on the infrastructure provided from PR #28564

@chsigg @whchung

### CODE REVIEW

File: tensorflow/core/util/gpu_cuda_alias.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Several forwarding macros are defined in this file to serve for backward
+// compatibility usage as we migrating from Cuda prefixed function to Gpu prefixed
```
Comment (chsigg) [inline tensorflow/core/util/gpu_cuda_alias.h]: Replace Cuda with CUDA in documentation (but keep Cuda in symbol names).

File: tensorflow/core/util/gpu_launch_config.h [hunk 2]
```diff
@@ -192,14 +192,7 @@ GpuLaunchConfig GetGpuLaunchConfig(int work_element_count,
...
 }
-template <typename DeviceFunc>
-CudaLaunchConfig GetCudaLaunchConfig(int work_element_count,
-                                     const Eigen::GpuDevice& d, DeviceFunc func,
-                                     size_t dynamic_shared_memory_size,
-                                     int block_size_limit) {
-  return GetGpuLaunchConfig(work_element_count, d, func,
-                            dynamic_shared_memory_size, block_size_limit);
-}
+CREATE_CUDA_HOST_FUNCTION_ALIAS(GetGpuLaunchConfig, GetCudaLaunchConfig);
```
Comment (chsigg) [inline tensorflow/core/util/gpu_launch_config.h]: I think you should #include "gpu_cuda_alias.h"

**Review by chsigg — CHANGES_REQUESTED** [review]
Thanks a lot, this is very easy to review.

File: tensorflow/core/util/gpu_kernel_helper.h [hunk 3]
```diff
@@ -53,16 +53,36 @@ limitations under the License.
...
 using gpuStream_t = hipStream_t;
 using gpuError_t = hipError_t;
 #endif
 
-#define GetGPUStream(context) context->eigen_gpu_device().stream()
-
 namespace tensorflow {
+#if GOOGLE_CUDA
+// cudaGetErrorString is available to both host and device
+__host__ __device__ inline const char* gpuGetErrorString(cudaError_t error){
```
Comment (chsigg) [inline tensorflow/core/util/gpu_kernel_helper.h]: This should be called 'GpuGetErrorString'. Maybe we can fix this in a follow-up PR?

**Review by chsigg — DISMISSED** [review]

File: tensorflow/core/util/gpu_kernel_helper.h (hunk 3 above)
Comment (jerryyin) [reply tensorflow/core/util/gpu_kernel_helper.h]: Oops another coding convention issue. Thanks for pointing out. This functions is rather isolated and has been applied to anywhere else. I'd rather file another commit that fix it now.

Looking at the CI result, I will add back the `GetCudaStream` function. I will file a follow up PR for making dependency changes from `GetCudaStream` to `GetGpuStream`.

File: tensorflow/core/util/gpu_launch_config.h [hunk 4]
```diff
@@ -387,19 +366,9 @@ Gpu2DLaunchConfig GetGpu2DLaunchConfig(int xdim, int ydim,
   return GetGpu3DLaunchConfig(xdim, ydim, 1, d, func,
                               dynamic_shared_memory_size, block_size_limit);
 }
+CREATE_CUDA_HOST_FUNCTION_ALIAS(GetGpu2DLaunchConfig, GetCuda2DLaunchConfig);
 
 #if GOOGLE_CUDA
-// Returns a raw reference to the current cuda stream.  Required by a
-// number of kernel calls (for which StreamInterface* does not work), i.e.
-// CUB and certain cublas primitives.
-inline const cudaStream_t& GetCudaStream(OpKernelContext* context) {
```
Comment (jerryyin) [inline tensorflow/core/util/gpu_launch_config.h]: @chsigg FYI, I was referring to here. Will add the function back, and file a follow-up PR to remove this as well as its dependencies

**Review by chsigg — APPROVED** [review]

**Pull request merged**